src/software/backend/simulation/simulator_robot_test.cpp

```diff
@@ -3,39 +3,688 @@
+};
+
+TEST_F(SimulatorRobotTest, test_robot_id)
+{
+    auto [world, simulator_robot, simulator_ball] =
+        createWorld(robot_non_zero_state, ball_zero_state);
+    EXPECT_EQ(simulator_robot->getRobotId(), 7);
+}
+
+TEST_F(SimulatorRobotTest, test_get_position)
+{
+    auto [world, simulator_robot, simulator_ball] =
+        createWorld(robot_non_zero_state, ball_zero_state);
+    EXPECT_FLOAT_EQ(simulator_robot->getPositionX(), 1.04);
+    EXPECT_FLOAT_EQ(simulator_robot->getPositionY(), -0.8);
+    EXPECT_LT((simulator_robot->position() - Point(1.04, -0.8)).length(), 0.001);
+}
+
+TEST_F(SimulatorRobotTest, test_get_orientation)
+{
+    auto [world, simulator_robot, simulator_ball] =
+        createWorld(robot_non_zero_state, ball_zero_state);
+    EXPECT_FLOAT_EQ(simulator_robot->getOrientation(), 2.12);
+}
+
+TEST_F(SimulatorRobotTest, test_get_linear_velocity)
+{
+    auto [world, simulator_robot, simulator_ball] =
+        createWorld(robot_non_zero_state, ball_zero_state);
+    EXPECT_NEAR(simulator_robot->getVelocityX(), -1.5, 0.01);
+    EXPECT_NEAR(simulator_robot->getVelocityY(), 0.0, 0.01);
+    EXPECT_LT((simulator_robot->velocity() - Vector(-1.5, 0)).length(), 0.01);
+}
+
+TEST_F(SimulatorRobotTest, test_get_angular_velocity)
+{
+    auto [world, simulator_robot, simulator_ball] =
+        createWorld(robot_non_zero_state, ball_zero_state);
+    EXPECT_FLOAT_EQ(simulator_robot->getVelocityAngular(), -1.0);
+}
+
+TEST_F(SimulatorRobotTest, test_get_battery_voltage)
+{
+    auto [world, simulator_robot, simulator_ball] =
+        createWorld(robot_non_zero_state, ball_zero_state);
+    EXPECT_FLOAT_EQ(simulator_robot->getBatteryVoltage(), 16.0);
+}
+
+TEST_F(SimulatorRobotTest, test_get_dribbler_temperature)
+{
+    auto [world, simulator_robot, simulator_ball] =
+        createWorld(robot_non_zero_state, ball_zero_state);
+    EXPECT_EQ(simulator_robot->getDribblerTemperatureDegC(), 25);
+}
+
+TEST_F(SimulatorRobotTest, test_kick)
+{
+    Robot robot(0, Point(0, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Point dribbling_point = getDribblingPoint(robot.position(), robot.orientation());
+    Ball ball(dribbling_point, Vector(0, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    // Simulate for 1/2 second without kicking
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    // Make sure we didn't kick
+    EXPECT_LT(simulator_ball->velocity().length(), 0.001);
+    EXPECT_LT((simulator_ball->position() - dribbling_point).length(), 0.01);
+
+    simulator_robot->kick(5.0);
+
+    // Simulate for 1/2 second after kicking
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(5, 0)).length(), 0.005);
+}
```

Comment:
Would be good to test this in a few directions

Reply:
done